train.py

```diff
@@ -1,115 +1,167 @@
+        self.rho = convertUnit(data['rho']['value'], data['rho']['unit'])  # rotating-mass factor [-]
 
-        self.rho = data['rho']  # rotating-mass factor [-]
+        if self.rho < 1:
 
-        # velocity limits
+            self.rho += 1  # 6% -> 0.06 -> 1.06
 
-        self.velocityMax = data['velocityMax']  # maximum train speed [m/s]
+        self.velocityMax = convertUnit(data['max speed']['value'], data['max speed']['unit'])  # maximum train speed [m/s]
 
-        # force limits
+        self.forceMax = convertUnit(data['max traction force']['value'], data['max traction force']['unit'])  # maximum traction force [N]
 
-        self.forceMax = data['forceMax']  # maximum traction force [N]
+        self.forceMin = convertUnit(-abs(data['max reg braking force']['value']), data['max reg braking force']['unit'])  # maximum regenerative braking force [N]
 
-        self.forceMin = data['forceMin']  # maximum regenerative braking force [N]
+        self.forceMinPn = convertUnit(-abs(data['max pn braking force']['value']), data['max pn braking force']['unit'])  # maximum pneumatic braking force [N]
 
-        self.forceMinPn = data['forceMinPn']  # maximum pneumatic braking force [N]
+        self.powerMax = convertUnit(data['max traction power']['value'], data['max traction power']['unit'])  # maximum traction power [W]
 
-        # power limits and losses
+        self.powerMin = convertUnit(-abs(data['max reg braking power']['value']), data['max reg braking power']['unit'])  # maximum regenerative braking power [W]
 
-        self.powerMax = data['powerMax']  # maximum traction power [W]
+        if 'max acceleration' in data:
 
-        self.powerMin = data['powerMin']  # maximum regenerative braking power [W]
+            self.accMax = convertUnit(data['max acceleration']['value'], data['max acceleration']['unit'])  # maximum acceleration [m/s^2]
 
-        self.powerLosses = lambda f,v: 0  # losses [W] as function of: force [N] and velocity [m/s]
+        if 'max deceleration' in data:
 
-        # acceleration/deceleration limits
+            self.accMin = convertUnit(-abs(data['max deceleration']['value']), data['max deceleration']['unit'])  # maximum allowed deceleration [m/s^2]
 
-        self.accMax = data['accMax']  # maximum acceleration [m/s^2]
+        self.r0 = convertUnit(data['rolling resistance r0']['value'], data['rolling resistance r0']['unit'])  # constant term [-]
 
-        self.accMin = data['accMin']  # maximum allowed deceleration [m/s^2]
+        self.r1 = convertUnit(data['rolling resistance r1']['value'], data['rolling resistance r1']['unit'])  # linear term [-]
 
-        # rolling resistance polynomial: f = r0 + r1*v + r2*v^2 [m/s -> N]
+        self.r2 = convertUnit(data['rolling resistance r2']['value'], data['rolling resistance r2']['unit'])  # quadratic term [-]
 
-        self.r0 = data['r0']  # constant term [-]
+        if 'efficiency traction' in data or 'efficiency reg brake' in data:
```

Comment:
Is it fine if both are omitted?

Reply:
if both omitted, we should properly check that dynamic efficiency is defined (added a todo). If a json file does not contain any information on efficiency, we should probably adopt some default values. To be discussed.